Add CLI page to Reference Section

ncapps opened the pull request:
This change consolidates `Commands` content on a single page to resolve https://github.com/kubernetes-sigs/kustomize/issues/5278.

I used the [kubectl CLI](https://kubernetes.io/docs/reference/kubectl/) page as inspiration for the content layout.

bugoverdose commented on site/content/en/docs/Reference/kustomize_cli.md:
```diff
@@ -1,8 +1,77 @@
 ---
-title: "kustomize CLI"
-linkTitle: "kustomize CLI"
+title: "Command line tool (kustomize)"
+linkTitle: "Command line tool (kustomize)"
```
I personlly think that the `tool` and `(kustomize)` part could be omitted.
The [kubectl CLI](https://kubernetes.io/docs/reference/kubectl/) page introduces a new tool called `kubectl` in the kubernetes Reference docs, so it's useful information. 
But right now we are already in the Kustomize document site, so it feels sort of redundant. 

```suggestion
title: "Command Line Interface"
linkTitle: "Command Line Interface"
```

<img width="1156" alt="image" src="https://github.com/kubernetes-sigs/kustomize/assets/73531614/ad104b40-8400-4087-af94-a2dd5072092d">

ncapps replied:
Updated title to "Command Line Interface"

bugoverdose reviewed (commented):
Just a minor comment.

bugoverdose commented on site/content/en/docs/Reference/kustomize_cli.md:
```diff
@@ -1,8 +1,77 @@
-linkTitle: "kustomize CLI"
+title: "Command line tool (kustomize)"
+linkTitle: "Command line tool (kustomize)"
 weight: 2
 date: 2017-01-05
```
I personally think that this page should go first, and then the API section should go next.
Right now CLI section's weight is 2 and API section's weight is 1.

```suggestion
weight: 0
date: 2017-01-05
```

bugoverdose commented on site/content/en/docs/Reference/kustomize_cli.md:
```diff
@@ -1,8 +1,77 @@
-linkTitle: "kustomize CLI"
+title: "Command line tool (kustomize)"
+linkTitle: "Command line tool (kustomize)"
 weight: 2
 date: 2017-01-05
```
I personally think that this page should go first, and then the API section should go next.
Right now CLI section's weight is 2 and API section's weight is 1.

```suggestion
weight: 0
date: 2017-01-05
```

ncapps replied:
Updated weight to 0.

bugoverdose commented on site/content/en/docs/Reference/kustomize_cli.md:
```diff
@@ -1,8 +1,77 @@
+
+Operation | Syntax | Description
+--- | --- | ---
+build | kustomize build DIR [flags] | Build a kustomization target from a directory or URL.
+cfg | kustomize cfg [command] | Commands for reading and writing configuration.
+completion | kustomize completion [bash\|zsh\|fish\|powershell] | Generate shell completion script.
+create | kustomize create [flags] | Create a new kustomization in the current directory.
+edit | kustomize edit [command] |  Edits a kustomization file.
+fn | kustomize fn [command] | Commands for running functions against configuration.
+localize | kustomize localize [target [destination]] [flags] | [Alpha] Creates localized copy of target kustomization root at destination.
+version | kustomize version [flags] | Prints the kustomize version.
```
How about doing it like this?

```suggestion
build | `kustomize build DIR [flags]` | Build a kustomization target from a directory or URL.
cfg | `kustomize cfg [command]` | Commands for reading and writing configuration.
completion | `kustomize completion [bash\|zsh\|fish\|powershell]` | Generate shell completion script.
create | `kustomize create [flags]` | Create a new kustomization in the current directory.
edit | `kustomize edit [command]` |  Edits a kustomization file.
fn | `kustomize fn [command]` | Commands for running functions against configuration.
localize | `kustomize localize [target [destination]] [flags]` | [Alpha] Creates localized copy of target kustomization root at destination.
version | `kustomize version [flags]` | Prints the kustomize version.
```

It would look like this:
<img width="775" alt="image" src="https://github.com/kubernetes-sigs/kustomize/assets/73531614/5648d40b-2f64-42a2-ba4f-75b683fb6198">

ncapps replied:
Added code syntax to markdown.

ncapps commented on site/content/en/docs/Reference/kustomize_cli.md:
```diff
@@ -1,8 +1,77 @@
----
\ No newline at end of file
+description:
+---
```
Thanks for catching this. I added a description.

ncapps commented:
Hi @bugoverdose, Thanks for your input! I updated everything per your comments.

natasha41575 commented:
/ok-to-test

ncapps commented:
Hi @bugoverdose, I believe that I addressed your comments. Would you please review when you get a chance? If you are good with the changes, would you please add a `/lgtm` comment?

bugoverdose reviewed (changes requested):
Hi sorry I was a bit late. Can you check the [Reference/CLI/_index.md](https://github.com/ncapps/kustomize/blob/docs/cli/site/content/en/docs/Reference/CLI/_index.md) page?

It was added from another PR, and I think we should migrate the content in `Reference/kustomize_cli.md` to `Reference/CLI/_index.md`

<img width="295" alt="image" src="https://github.com/kubernetes-sigs/kustomize/assets/73531614/923015c4-55cc-4ce8-bbe0-d624eb4b38c2">

ncapps commented:
> Hi sorry I was a bit late. Can you check the [Reference/CLI/_index.md](https://github.com/ncapps/kustomize/blob/docs/cli/site/content/en/docs/Reference/CLI/_index.md) page?
> 
> It was added from another PR, and I think we should migrate the content in `Reference/kustomize_cli.md` to `Reference/CLI/_index.md`
> 
> <img alt="image" width="295" src="https://user-images.githubusercontent.com/73531614/267847882-923015c4-55cc-4ce8-bbe0-d624eb4b38c2.png">

@bugoverdose Thank you for pointing this out. I moved the content to the CLI index. 
<img width="247" alt="Screenshot 2023-09-15 at 7 42 03 AM" src="https://github.com/kubernetes-sigs/kustomize/assets/10539313/08ad3785-f191-458c-92e3-92ca80c8e9ea">

bugoverdose commented:
/lgtm

ncapps commented:
/label tide/merge-method-squash

annasong20 commented:
Looks great, thanks @ncapps! Taking an iterative approach, anyone who wants more detail can add it later.
/approve
